CMakeLists.txt

```diff
@@ -261,11 +256,65 @@ include(cmake/leveldb.cmake)
 include(cmake/minisketch.cmake)
 include(cmake/secp256k1.cmake)
 
+string(STRIP "${CMAKE_CXX_FLAGS}" cxx_flags)
+string(STRIP "${CMAKE_CXX_FLAGS_INIT}" cxx_flags_init)
+if(cxx_flags STREQUAL cxx_flags_init)
```

Comment:
Non-blocker IMO, just mentioning: with this change it is no longer possible to set your own flags on the command line like `cmake -DCMAKE_CXX_FLAGS=-myflag` and have stuff from `try_append_cxx_flags()` from inside `CMakeLists.txt` appended to them.

To achieve this either one of the following worked for me:

`export CXXFLAGS="-myflag"` (and then run `cmake`)
or
`-DCMAKE_CXX_COMPILER="c++;-myflag"`

Reply:
> -DCMAKE_CXX_COMPILER="c++;-myflag"

This is definitely not someting that we would want (or encourage others) to do.